### PR TITLE
JAMES-2375 DSNBounce mailet do not provide a subject

### PR DESCRIPTION
**Pull request by splainez**
Subject empty fixed, Sender empty when you do not provide sender
property fixed too

https://issues.apache.org/jira/browse/JAMES-2375

### CODE REVIEW

**Comment by chibenwa** [comment]
Thanks you very much for your contribution on that topic!

I will take a look at it as soon as I can.

In the meantime, I scheduled tests on it: https://github.com/linagora/james-project/pull/2673

Cheers

Benoit

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java [hunk 1]
```diff
@@ -523,7 +526,121 @@ public void dsnBounceShouldAddPrefixToSubjectWhenPrefixIsConfigured() throws Exc
                 .build();
 
         dsnBounce.service(mail);
+        
 
-        assertThat(mail.getMessage().getSubject()).isEqualTo("pre My subject");
+        List<SentMail> sentMails = fakeMailContext.getSentMails();
+        assertThat(sentMails).hasSize(1);
+        SentMail sentMail = sentMails.get(0);
+        assertThat(sentMail.getSubject().orElse(null)).isEqualTo("pre My subject");
```
Comment (chibenwa) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java]: You can rely on list + optional asserts:

```
assertThat(fakeMailContext.getSentMails())
    .hasSize(1)
    .allSatisfies(mail -> assertThat(mail.getSubject()).contains("...));
```

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java [hunk 2]
```diff
@@ -523,7 +526,119 @@ public void dsnBounceShouldAddPrefixToSubjectWhenPrefixIsConfigured() throws Exc
                 .build();
 
         dsnBounce.service(mail);
+        
+        assertThat(fakeMailContext.getSentMails())
+        .hasSize(1)
+        .allSatisfy(sentMail -> assertThat(sentMail.getSubject()).contains("pre My subject"));
```
Comment (Arsnael) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java]: Missing indent

Comment (splainez) [reply server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java]: fixed

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java [hunk 3]
```diff
@@ -454,6 +453,6 @@ private MimeBodyPart createAttachedOriginal(Mail originalMail, TypeCode attachme
 
     @Override
     public MimeMessageModifier getMimeMessageModifier(Mail newMail, Mail originalMail) throws MessagingException {
-        return new MimeMessageModifier(originalMail.getMessage());
+        return new MimeMessageModifier(newMail.getMessage());
```
Comment (remk) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java]: Why ?

Comment (splainez) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java]: This method is used to add the prefix to the subject of the message, so the modification is to the new email and not to the original email.

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

One very last thing: I don't have write access to this repository, hence I can not close this issue. Wouldn't you mind doing it for me? It would avoid me annoying the Apache INFRA team with such simple concerns.

Thanks again,

Cheers,

Benoit

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java (hunk 3 above)
Comment (chibenwa) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java]: Thanks for the explanation!

**Comment by splainez** [comment]
merged https://github.com/linagora/james-project/pull/2673

**Pull request closed without merging**